src/Workspaces/VisualBasic/Portable/Extensions/CastAnalyzer.vb

```diff
@@ -217,7 +217,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions
 
             ' A casts to object can always be removed from an expression inside of an interpolation, since it'll be converted to object
             ' in order to call string.Format(...) anyway.
-            If castType?.SpecialType = SpecialType.System_Object AndAlso
+            If (castType?.SpecialType = SpecialType.System_Object).GetValueOrDefault() AndAlso
```

Comment:
>(castType?.SpecialType = SpecialType.System_Object).GetValueOrDefault() [](start = 15, length = 71)

This is an optimization

Reply:
I'm confused: is this ensuring the same optimization happens once Roslyn is built with a newer version of itself after this PR is merged? We absolutely don't care about that in the IDE cases, so we should just keep the code simple.

Reply:
>  is this ensuring the same optimization happens once Roslyn is built with a newer version of itself after this PR is merged?

Yes.

> We absolutely don't care about that in the IDE cases, so we should just keep the code simple.

The new code looks simple to me. If you insist, I can revert it.

---
In reply to: [327884085](https://github.com/dotnet/roslyn/pull/38802#discussion_r327884085) [](ancestors = 327884085)

Reply:
Ah, I misunderstood -- didn't realize the null case continues to evaluate.